libavcodec/qsvenc.c

```diff
@@ -1304,12 +1312,18 @@ static int submit_frame(QSVEncContext *q, const AVFrame *frame,
 
         qf->surface = *(mfxFrameSurface1*)qf->frame->data[3];
 
+
         if (q->frames_ctx.mids) {
             ret = ff_qsv_find_surface_idx(&q->frames_ctx, qf);
-            if (ret < 0)
-                return ret;
-
-            qf->surface.Data.MemId = &q->frames_ctx.mids[ret];
+            if (ret >= 0)
+                qf->surface.Data.MemId = &q->frames_ctx.mids[ret];
+        }
+        if (!q->frames_ctx.mids || ret < 0) {
```

Comment:
if q->frames_ctx.mids is not NULL, ret < 0 is a error.

Reply:
if mids is not NULL, ret is the return value from ff_qsv_find_surface_idx(), so if ret >=0 we skip this branch, and if ret < 0, it means the surface is not in the pool, and we need go into this branch to create new resource for this frame.

Reply:
in some error cases, mids are not null, and ret will < 0 too. In this case we can't create a new resource for it, we need to report an error.

Reply:
Ok, I got your point.
The ff_qsv_find_surface_idx() is a simple function, It returns minus value only when the surface is not found, so I don't category the returned value.